File: Quiz 1/Palak_Rastogi_590018052/Quiz_1_Debugging_Question_1.py
'''
Bug : the code resets currentSum to 0 when it's negative
'''

#Corrected Code


def max_subarray_sum(nums):
    # Initialize both to the first element
    max_sum = current_sum = nums[0]

    # Start loop from the second element
    for i in range(1, len(nums)):
        current_sum = max(nums[i], current_sum + nums[i])
        max_sum = max(max_sum, current_sum)

    return max_sum

# Example usage
arr = [-3, -2, -1, -4]
result = max_subarray_sum(arr)
print("Maximum subarray sum is:", result)

File: Quiz_1/Shreya_Goyal_590015491/Quiz_1_Debugging_Question_1.c
#include <stdio.h>

int maxSubarraySum(int nums[], int n)
{
    int maxSum = nums[0];
    int currentSum = 0;

    for (int i = 1; i < n; i++) 
   {
        currentSum += nums[i];
        if (currentSum < 0) 
       {
            currentSum = 0;
        }
        if (currentSum < maxSum) 
       {
            currentSum = nums[i];
        }
    }
    return maxSum;

}